src/SmartComponents/BaselinesTable/BaselinesTable.js

```diff
@@ -30,19 +30,17 @@ class BaselinesTable extends Component {
 
     onSelect(event, isSelected, rowId) {
         const { baselineTableData, selectBaseline } = this.props;
-        let rows;
+        let ids;
 
         if (rowId === -1) {
-            rows = baselineTableData.map(oneRow => {
-                oneRow.selected = isSelected;
-                return oneRow;
+            ids = baselineTableData.map(function(item) {
+                return item[0];
             });
         } else {
-            rows = [ ...baselineTableData ];
-            rows[rowId].selected = isSelected;
+            ids = [ baselineTableData[rowId][0] ];
         }
 
-        selectBaseline(rows);
+        selectBaseline(ids, isSelected, baselineTableData);
```

Comment:
Here you're passing 3 arguments to selectBaseline, but in the actions, you are only accepting 2.